vsphere/resource_vsphere_vnic.go

```diff
@@ -0,0 +1,517 @@
+package vsphere
+
+import (
+	"context"
+	"fmt"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/terraform-providers/terraform-provider-vsphere/vsphere/internal/helper/hostsystem"
+	"github.com/vmware/govmomi"
+	"github.com/vmware/govmomi/object"
+	"github.com/vmware/govmomi/vim25/mo"
+	"github.com/vmware/govmomi/vim25/types"
+	"log"
+	"strconv"
+	"strings"
+)
+
+func resourceVsphereNic() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceVsphereNicCreate,
+		Read:   resourceVsphereNicRead,
+		Update: resourceVsphereNicUpdate,
+		Delete: resourceVsphereNicDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: vNicSchema(),
+	}
+}
+
+func vNicSchema() map[string]*schema.Schema {
+	base := BaseVMKernelSchema()
+	base["host"] = &schema.Schema{
+		Type:        schema.TypeString,
+		Required:    true,
+		Description: "ESX host the interface belongs to",
+		ForceNew:    true,
+	}
+
+	return base
+}
+
+func resourceVsphereNicRead(d *schema.ResourceData, meta interface{}) error {
+	ctx := context.TODO()
+	client := meta.(*VSphereClient).vimClient
+	tfNicID := d.Id()
+
+	toks := strings.Split(tfNicID, "_")
+	hostId := toks[0]
+	nicId := toks[1]
+
+	vnic, err := getVnicFromHost(ctx, client, hostId, nicId)
+	if err != nil {
+		log.Printf("[DEBUG] Nic (%s) not found. Probably deleted.", nicId)
+		d.SetId("")
+		return nil
+	}
+
+	_ = d.Set("portgroup", vnic.Portgroup)
+	if vnic.Spec.DistributedVirtualPort != nil {
+		_ = d.Set("distributed_switch_port", vnic.Spec.DistributedVirtualPort.SwitchUuid)
+		_ = d.Set("distributed_port_group", vnic.Spec.DistributedVirtualPort.PortgroupKey)
+	}
+	_ = d.Set("mtu", vnic.Spec.Mtu)
+	_ = d.Set("mac", vnic.Spec.Mac)
+	//_ = d.Set("netstack", vnic.Spec.NetStackInstanceKey)
```

Comment:
If this isn't needed in the current implementation or as a future reminder, it should be removed to clean up the clutter.